apps/src/blockly/addons/cdoSerializationHelpers.js

```diff
@@ -48,7 +48,7 @@ export function addPositionsToState(xmlBlocks, blockIdMap) {
   xmlBlocks.forEach(xmlBlock => {
     const blockJson = blockIdMap.get(xmlBlock.blockly_block.id);
     if (blockJson) {
-      // Do not change values if xmlBlock values are NaN (unspecified in XML)
+      // Note: If xmlBlock values are NaN, they will be ignored and blockJson values will be used
```

Comment:
I was thinking about when this regression was introduced - based on your description, it seems like it has to do with the call to `getRelativeToSurfaceXY` within `isBlockLocationUnset`. It's interesting that this was added ~4 months ago by https://github.com/code-dot-org/code-dot-org/pull/52072 and nobody noticed the regression until recently. Do you think there's another factor involved - just trying to understand the bug better. Thanks!

Reply:
I'm not 100% sure, but I am happy to checkout an older branch and update this comment! I think https://github.com/code-dot-org/code-dot-org/pull/52072 was an experiment until https://github.com/code-dot-org/code-dot-org/pull/52222 (~2 months ago instead of ~4 months), so maybe that was a factor?

Reply:
Ah..that would make more sense if it's only been a couple months outside of the experiment. Thanks!